[data] add data dashboard docs (#40367)
ray-project/ray: doc/source/data/performance-tips.rst

> Monitoring your application

Reviewer: In the future, let's probably move this section out as a separate page, because this is important.

---------------------------

> View the Ray Data dashboard located in the Metrics tab of the Ray Dashboard to monitor your application and troubleshoot issues. Ray Data emits Prometheus metrics in real-time while a Dataset is executing, and the Ray Data dashboard displays these metrics grouped by Dataset. Datasets can also be assigned a name using :meth:`~ray.data.Dataset._set_name`, which prefixes the dataset id for a more identifiable label.

Reviewer 2: ```suggestion
View the Ray Data dashboard located in the Metrics tab of the Ray Dashboard to monitor your application and troubleshoot issues. Ray Data emits Prometheus metrics in real-time while a Dataset is executing, and the Ray Data dashboard displays these metrics grouped by Dataset. Datasets can also be assigned a name using :meth:`~ray.data.Dataset._set_name`, which prefixes the dataset ID for a more identifiable label.
```

> View the Ray Data dashboard located in the Metrics tab of the Ray Dashboard to monitor your application and troubleshoot issues. Ray Data emits Prometheus metrics in real-time while a Dataset is executing, and the Ray Data dashboard displays these metrics grouped by Dataset. Datasets can also be assigned a name using :meth:`~ray.data.Dataset._set_name`, which prefixes the dataset ID for a more identifiable label.

Reviewer: looks like it just shows `_set_name` on the UI, can we change it to `Dataset._set_name`?

<img width="965" alt="Screen Shot 2023-10-20 at 11 43 20 AM" src="https://github.com/ray-project/ray/assets/4629931/ea3a5712-3d0e-42ca-9aa5-db7ef9a09eed">

Reviewer: >located in the Metrics tab of the Ray Dashboard

Can we add a link to https://docs.ray.io/en/latest/ray-observability/getting-started.html#metrics-view ? So users are clear what we are referring to.

The metrics recorded are:

* Bytes Spilled
* Bytes Allocated
* Bytes Freed
* Object Store Memory
* CPUs (logical slots)
* GPUs (logical slots)

> * Bytes Outputted

Reviewer: can we be more verbose for the metrics?

e.g.
`Bytes Spilled` -> `Objects in Bytes Spilled from Object Store to Disk`

> * Bytes of objecst freed in object store

Reviewer: `objecst` -> `objects`. wow nice, Vale caught this issue, @bveeramani.

Monitoring your application
---------------------------

> View the Ray Data dashboard located in the Metrics tab of the Ray Dashboard to monitor your application and troubleshoot issues. Ray Data exports metrics in real-time while a Dataset is executing, and the Ray Data dashboard displays these metrics grouped by Dataset.

Reviewer 3: ```suggestion
View the Ray Data dashboard located in the Metrics tab of the Ray Dashboard to monitor your application and troubleshoot issues. Ray Data emits metrics via Prometheus in real-time while a Dataset is executing, and the Ray Data dashboard displays these metrics grouped by Dataset.
```

> To learn more about the Ray dashboard, see :ref:`Ray Dashboard <observability-getting-started>`.

Reviewer 3: ```suggestion
To learn more about the Ray dashboard, including detailed setup instructions, see :ref:`Ray Dashboard <observability-getting-started>`.
```
